__tests__/app.test.js

```diff
@@ -579,3 +523,36 @@ describe("GET /api/users", () => {
       });
   });
 });
+
+describe("GET /api/articles (topic query)", () => {
+  test("response code is 200", () => {
+    return request(app).get("/api/articles?topic=cats").expect(200);
+  });
+  test("returns an array of correct length", () => {
+    return request(app)
+      .get("/api/articles?topic=cats")
+      .expect(200)
+      .then((response) => {
+        const result = response.body;
+        expect(result.length).toBe(1);
+      });
+  });
+  test("returns error if given an invalid topic query", () => {
+    return request(app)
+      .get("/api/articles?topic=dogs")
+      .expect(400)
+      .then((response) => {
+        const result = response.body.msg;
+        expect(result).toBe("Bad request");
+      });
+  });
+  test("returned article has a topic matching the requested topic", () => {
+    return request(app)
+      .get("/api/articles?topic=cats")
+      .expect(200)
+      .then((response) => {
+        const result = response.body[0].topic;
+        expect(result).toBe("cats");
+      });
+  });
```

Comment:
What about a valid topic that has no articles? e.g. ?topic=paper from in the test data

(we'll need to differentiate between a topic that does exist, but has no articles (200 OK) vs a topic that doesn't exist at all (404 not found))

We can consider using another query to the topics db to see if the requested topic exists or not